model/src/main/proto/spine/validation/rule.proto

```diff
@@ -0,0 +1,164 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+syntax = "proto3";
+
+package spine.validation;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.validation";
+option java_outer_classname = "RuleProto";
+option java_multiple_files = true;
+
+import "spine/validation/value.proto";
+import "spine/protodata/ast.proto";
+import "google/protobuf/any.proto";
+
+// A simple validation rule.
+//
+// May be expressed as a comparison between two operands. The first operand is always a field value.
+// The second operand is the `other_value`. `other_value` is typically derived from the Protobuf
+// definition of the validation rule.
+//
+message SimpleRule {
+
+    // The field which is being compared.
+    protodata.Field field = 1;
+
+    oneof operation {
+
+        // A simple comparison sign.
+        ComparisonOperator sign = 2;
+
+        // A custom comparison method, which cannot be covered by a combination of `Sign`s.
+        CustomOperator custom_operator = 3;
+    }
+
+    // The value to which the field is compared.
+    Value other_value = 4;
+
+    // A human-readable message explaining the error.
+    //
+    // May include tokens `{value}` and `{other}`, which will be replaced with the runtime values
+    // when the error is constructed.
+    //
+    string error_message = 5;
+}
+
+// A predefined comparison operator.
+//
+// Comparison between two values of the same type must always yield a boolean result.
+//
+enum ComparisonOperator {
+
+    OPERATION_UNKNOWN = 0;
+
+    EQUAL = 1;
+
+    NOT_EQUAL = 2;
+
+    LESS_THAN = 3;
+
+    GREATER_THAN = 4;
+
+    LESS_OR_EQUAL = 5;
+
+    GREATER_OR_EQUAL = 6;
+
+    DISTINCT = 7;
+}
+
+// A non-predefined comparison operator.
+message CustomOperator {
+
+    // Describes how the renderer should compare the values.
+    string description = 1;
```

Comment:
What if we add a field `name` before the `description`? It would be better to have a short form text _and_ a longer one with documentation. It would help formulating both.

Reply:
This type is likely to change. 

At this moment, we simply ignore any custom operators. We also, of course, do not generate them.

In future, we might need some kind of a way to inject renderers with non-predefined operations. I assume we might simply put code expressions in target languages in there.

In this PR, I'd like to leave this type alone and see how it goes when we add more validation rules into the mix. The structure of a `CustomOperator` is unknown for now.